tests/e2e/send-message.js

```diff
@@ -54,319 +56,195 @@ describe('Send message', () => {
     parent: { _id: DAVID_AREA._id },
   };
 
+  const messageDoc = {
+    '_id': uuid(),
+    'errors': [],
+    'form': null,
+    'reported_date': new Date().getTime(),
+    'tasks': [
+      {
+        'messages': [
+          {
+            'sent_by': 'admin',
+            'to': '+447765902000',
+            'message': 'Hello raw this is a test SMS',
+            'uuid': uuid()
+          }
+        ],
+        'state_history': [
+          {
+            'state': 'pending',
+            'timestamp': '2021-02-12T17:34:47.734Z'
+          }
+        ],
+        'state': 'pending'
+      }
+    ],
+    'kujua_message': true,
+    'type': 'data_record',
+    'sent_by': 'admin'
+  };
+
+  const aliceMessageDoc = _.cloneDeep(messageDoc);
+  aliceMessageDoc.tasks[0].messages[0] = {
+    'sent_by': 'admin',
+    'to': '+447765902001',
+    'contact': {
+      '_id': 'alice-contact'
+    },
+    'message': 'Hello alice-contact this is a test SMS',
+    'uuid': uuid()
+  };
+
   const CONTACTS = [ALICE, BOB_PLACE, CAROL, JAROL, DAVID, DAVID_AREA];
 
-  beforeAll(done => {
+  beforeEach(async () => {
     DAVID_AREA.contact = { _id: DAVID._id, phone: '+447765902002' };
-    protractor.promise
-      .all(CONTACTS.map(utils.saveDoc))
-      .then(done)
-      .catch(done.fail);
+    await utils.saveDocs(CONTACTS);
```

Comment:
These tests were very much dependent on each other.  I modified this so they can run individually otherwise they fail because when I run a single test it expects there to be other messages that don't exist because the prior test didn't run.

Reply:
Ok cool, I agree with that!